src/components/__tests__/collapsible.test.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import { render, act, fireEvent } from "@testing-library/react"
+
+import Collapsible from "../collapsible"
+
+const openedText = "Close me"
+const closedText = "Open me"
+const content = "I am content"
+
+const renderWrapper = ({ isInitiallyCollapsed = true, onChange = null } = {}) =>
+  render(
+    <Collapsible
+      renderToggle={(isCollapsed) => (isCollapsed ? closedText : openedText)}
+      onChange={onChange}
+      isInitiallyCollapsed={isInitiallyCollapsed}
+    >
+      {() => content}
+    </Collapsible>
+  )
+
+describe("<Collapsible>", () => {
+  describe("default", () => {
+    it("is closed", () => {
+      const { getByText, queryAllByText } = renderWrapper()
+
+      expect(getByText(closedText))
+      expect(queryAllByText(openedText)).toEqual([])
+      expect(queryAllByText(content)).toEqual([])
+    })
+
+    it("can be opened", () => {
+      const { getByText, queryAllByText } = renderWrapper()
+
+      act(() => {
+        fireEvent.click(getByText(closedText))
+      })
+
+      expect(queryAllByText(closedText)).toEqual([])
+      expect(getByText(openedText))
+      expect(getByText(content))
+    })
+  })
+
+  describe("rendered open", () => {
+    it("is open", () => {
+      const { getByText, queryAllByText } = renderWrapper({
+        isInitiallyCollapsed: false,
+      })
+
+      expect(queryAllByText(closedText)).toEqual([])
+      expect(getByText(openedText))
+      expect(getByText(content))
+    })
+
+    it("can be closed", () => {
+      const { getByText, queryAllByText } = renderWrapper({
+        isInitiallyCollapsed: false,
+      })
+
+      act(() => {
+        fireEvent.click(getByText(openedText))
+      })
+
+      expect(getByText(closedText))
+      expect(queryAllByText(openedText)).toEqual([])
+      expect(queryAllByText(content)).toEqual([])
+    })
+  })
+
+  describe("on change function", () => {
+    it("should call the onChange function if defined", () => {
+      const customOnChangeFunction = jest.fn()
+      const { getByText } = renderWrapper({
+        onChange: customOnChangeFunction,
+      })
+
+      act(() => {
```

Comment:
I think to remember that most of the react-testing-library helpers are wrapped with `act` internally and most of the time, we don't need to `act` in the test ourselves. can you confirm why we need to to do explicitly here?

Reply:
no we don't. This was before my deep dive into the testing library during the holidays. `.click()` already wraps everything into act. Removed it